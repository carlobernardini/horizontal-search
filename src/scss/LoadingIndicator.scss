@import 'base';

.orizzonte__loading {
    animation: rotate 2s linear infinite;
    width: 65px;
    height: 65px;
    display: block;
}

.orizzonte__loading-path {
    stroke-dasharray: 1,150; /* 1%, 101% circumference */
    stroke-dashoffset: 0;
    stroke: $loading-indicator-color;
    stroke-linecap: round;
    animation: dash 1.5s ease-in-out infinite;
}
