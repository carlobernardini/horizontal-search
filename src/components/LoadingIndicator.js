import React from 'react';
import PropTypes from 'prop-types';
import '../scss/LoadingIndicator.scss';

const LoadingIndicator = ({ size }) => (
    <svg
        className="orizzonte__loading"
        viewBox={ [0, 0, 44, 44] }
        style={{
            width: size,
            height: size
        }}
    >
        <circle
            className="orizzonte__loading-path"
            cx="22"
            cy="22"
            r="20"
            fill="none"
            strokeWidth="4"
        />
    </svg>
);

LoadingIndicator.displayName = 'OrizzonteLoadingIndicator';

LoadingIndicator.propTypes = {
    size: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string
    ])
};

LoadingIndicator.defaultProps = {
    size: null
};

export default LoadingIndicator;
