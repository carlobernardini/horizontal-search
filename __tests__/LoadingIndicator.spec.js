import React from 'react';
import LoadingIndicator from '../src/components/LoadingIndicator';

describe('<LoadingIndicator />', () => {
    it('should render a default loading indicator', () => {
        const wrapper = shallow(
            <LoadingIndicator />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render a custom loading indicator', () => {
        const wrapper = shallow(
            <LoadingIndicator
                size="12"
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
