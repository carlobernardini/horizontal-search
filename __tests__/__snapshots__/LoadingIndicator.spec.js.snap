// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should render a custom loading indicator 1`] = `
<svg
  className="orizzonte__loading"
  style={
    Object {
      "height": "12",
      "width": "12",
    }
  }
  viewBox={
    Array [
      0,
      0,
      44,
      44,
    ]
  }
>
  <circle
    className="orizzonte__loading-path"
    cx="22"
    cy="22"
    fill="none"
    r="20"
    strokeWidth={4}
  />
</svg>
`;

exports[`<LoadingIndicator /> should render a default loading indicator 1`] = `
<svg
  className="orizzonte__loading"
  style={
    Object {
      "height": null,
      "width": null,
    }
  }
  viewBox={
    Array [
      0,
      0,
      44,
      44,
    ]
  }
>
  <circle
    className="orizzonte__loading-path"
    cx="22"
    cy="22"
    fill="none"
    r="20"
    strokeWidth={4}
  />
</svg>
`;
